test/e2e/pkg/grammar/abci_grammar.md

```diff
@@ -29,32 +31,39 @@ ProcessProposal : "process_proposal" ;
 
 ```
 
-The part of the original grammar (https://github.com/cometbft/cometbft/blob/main/spec/abci/abci%2B%2B_comet_expected_behavior.md) the grammar above 
+The original grammar (https://github.com/cometbft/cometbft/blob/main/spec/abci/abci%2B%2B_comet_expected_behavior.md) the grammar above 
 refers to is below: 
 
-start               = clean-start
+start               = clean-start / recovery
 
-clean-start         = init-chain [state-sync] consensus-exec
+clean-start         = ( app-handshake / state-sync ) consensus-exec
+app-handshake       = info init-chain
 state-sync          = *state-sync-attempt success-sync info
 state-sync-attempt  = offer-snapshot *apply-chunk
 success-sync        = offer-snapshot 1*apply-chunk
 
+recovery            = info [init-chain] consensus-exec
+
 consensus-exec      = (inf)consensus-height
-consensus-height    = *consensus-round decide commit
+consensus-height    = *consensus-round finalize-block commit
 consensus-round     = proposer / non-proposer
 
-proposer            = [prepare-proposal [process-proposal]]
-non-proposer        = [process-proposal] 
+proposer            = *got-vote [prepare-proposal [process-proposal]] [extend]
+extend              = *got-vote extend-vote *got-vote
+non-proposer        = *got-vote [process-proposal] [extend]
 
 init-chain          = %s"<InitChain>"
-decide              = %s"<FinalizeBlock>"
-commit              = %s"<Commit>"
 offer-snapshot      = %s"<OfferSnapshot>"
 apply-chunk         = %s"<ApplySnapshotChunk>"
 info                = %s"<Info>"
 prepare-proposal    = %s"<PrepareProposal>"
 process-proposal    = %s"<ProcessProposal>"
+extend-vote         = %s"<ExtendVote>"
+got-vote            = %s"<VerifyVoteExtension>"
+finalize-block      = %s"<FinalizeBlock>"
+commit              = %s"<Commit>"
 
+*Note* `Info` and parts related to vote extensions are missing. 
```

Comment:
Could you add a sentence briefly explaning that `Info` is missing because we're ignoring it in the tests, due to the fact that it is also triggered by the e2e tests at unpredictable places?

Reply:
I will do this!